### PR TITLE
Enable availability zone feature

### LINKED ISSUE #7: Add more e2e test cases
Azure related features should be tested in e2e tests, which includes

- [ ] Different authz methods
- [ ] Different load balancer behavior with annotations
- [ ] Verification of various resources
  - [ ] Routes
  - [ ] Node's externalID
  - [ ] Node's publicIP
  - [ ] NSGs
- [ ] ACR image pulling tests without docker secrets setting explicitly
- [ ] Multi availability zones tests
- [ ] Cloud provider configuration options tests (e.g. Standard LoadBalancer and VMSS)
- [ ] Upgrading tests (e.g. cloud-provider version updates)

Also add tests for Azure persistent storages, e.g.

- [ ] AzureFile
- [ ] AzureDisk

### PR DESCRIPTION
**Pull request by Abhik1998**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
kind feature
> /kind flake

**What this PR does / why we need it**:
Add availability zone feature.
**Which issue(s) this PR fixes**:

Fixes #7 

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: hack/deploy-cluster.sh [hunk 1]
```diff
@@ -17,30 +17,46 @@ set -e
 REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
 RESOURCE_GROUP_NAME=${RESOURCE_GROUP_NAME:-""}
 LOCATION=${LOCATION:-""}
+
+
+#check for variables is initialized or not
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: nit: remove empty lines

File: hack/deploy-cluster.sh [hunk 2]
```diff
@@ -17,30 +17,46 @@ set -e
...
 TENANT_ID=${TENANT_ID:-""}
 USE_CSI_DEFAULT_STORAGECLASS=${USE_CSI_DEFAULT_STORAGECLASS:-""}
+ENABLE_AVAILABILITY_ZONE=${ENABLE_AVAILABILITY_ZONE:-""}
 
-# Check for variables is initialized or not
-if [ -z "$LOCATION" ] || [ -z "${SUBSCRIPTION_ID}" ] || [ -z "${CLIENT_ID}" ] || [ -z "${CLIENT_SECRET}" ] || [ -z "${TENANT_ID}" ] || [ -z "${USE_CSI_DEFAULT_STORAGECLASS}" ]; then
-  echo "SUBSCRIPTION_ID, CLIENT_ID, TENANT_ID, CLIENT_SECRET ,LOCATION and USE_CSI_DEFAULT_STORAGECLASS must be specified"
+# check for variables is initialized or not
+if [ -z "$LOCATION" ] || [ -z "${SUBSCRIPTION_ID}" ] || [ -z "${CLIENT_ID}" ] || [ -z "${CLIENT_SECRET}" ] || [ -z "${TENANT_ID}" ] || [ -z "${USE_CSI_DEFAULT_STORAGECLASS}" ] || [ -z "${ENABLE_AVAILABILITY_ZONE}"]; then
+  echo "SUBSCRIPTION_ID, CLIENT_ID, TENANT_ID, CLIENT_SECRET ,LOCATION ,USE_CSI_DEFAULT_STORAGECLASS and ENABLE_AVAILABILITY_ZONE must be specified"
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: let's make ENABLE_AVAILABILITY_ZONE  optional. When ENABLE_AVAILABILITY_ZONE  is not set, disable the AZ

File: hack/deploy-cluster.sh [hunk 3]
```diff
@@ -17,30 +17,46 @@ set -e
...
 fi
 
-
 if [ "$RESOURCE_GROUP_NAME" = "" ]
 then
 echo "RESOURCE GROUP NAME must be specified"
 exit 1
+
+
+# check for variables is initialized or not
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: why check again here?

File: hack/deploy-cluster.sh [hunk 4]
```diff
@@ -50,6 +66,122 @@ if ! [ -x "$(command -v aks-engine)" ]; then
   exit 1
 fi
 
+# initialize variables
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: this added too much duplicated codes, please remove those lines.

Let's define a variable for `${REPO_ROOT}/examples/aks-engine.json`. When AZ is enabled, change the value to `${REPO_ROOT}/examples/az.json`

File: hack/deploy-cluster.sh (hunk 2 above)
Comment (Abhik1998) [reply hack/deploy-cluster.sh]: okay , to set that the user can put ENABLE_AVAILABILITY_ZONE=yes

File: hack/deploy-cluster.sh (hunk 4 above)
Comment (Abhik1998) [reply hack/deploy-cluster.sh]: Tried to do the same by defining a variable root_file=az.json

File: hack/deploy-cluster.sh [hunk 5]
```diff
@@ -17,30 +17,31 @@ set -e
...
+#check for variables is initialized or not
 SUBSCRIPTION_ID=${SUBSCRIPTION_ID:-""}
 CLIENT_ID=${CLIENT_ID:-""}
 CLIENT_SECRET=${CLIENT_SECRET:-""}
 TENANT_ID=${TENANT_ID:-""}
 USE_CSI_DEFAULT_STORAGECLASS=${USE_CSI_DEFAULT_STORAGECLASS:-""}
+ENABLE_AVAILABILITY_ZONE=${ENABLE_AVAILABILITY_ZONE:-""}
 
-# Check for variables is initialized or not
+# check for variables is initialized or not
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: Let's keep the capital here and below

File: hack/deploy-cluster.sh [hunk 6]
```diff
@@ -17,30 +17,31 @@ set -e
...
 if [ -z "$LOCATION" ] || [ -z "${SUBSCRIPTION_ID}" ] || [ -z "${CLIENT_ID}" ] || [ -z "${CLIENT_SECRET}" ] || [ -z "${TENANT_ID}" ] || [ -z "${USE_CSI_DEFAULT_STORAGECLASS}" ]; then
-  echo "SUBSCRIPTION_ID, CLIENT_ID, TENANT_ID, CLIENT_SECRET ,LOCATION and USE_CSI_DEFAULT_STORAGECLASS must be specified"
+  echo "SUBSCRIPTION_ID, CLIENT_ID, TENANT_ID, CLIENT_SECRET ,LOCATION ,USE_CSI_DEFAULT_STORAGECLASS and ENABLE_AVAILABILITY_ZONE must be specified"
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: could you remove ENABLE_AVAILABILITY_ZONE from here, since it is optional?

File: hack/deploy-cluster.sh [hunk 7]
```diff
@@ -62,9 +63,14 @@ function cleanup() {
   rm -f ${manifest_file}
 }
 trap cleanup EXIT
-
+if [ -z "$ENABLE_AVAILABILITY_ZONE" ]
+then
+ base_manifest_file=${REPO_ROOT}/examples/aks-engine.json
+else
+ base_manifest_file=${REPO_ROOT}/examples/az.json
+fi
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: change this to 

```
base_manifest_file=${REPO_ROOT}/examples/aks-engine.json
if [ ! -z "$ENABLE_AVAILABILITY_ZONE" ]; then
    base_manifest_file=${REPO_ROOT}/examples/az.json
fi

File: hack/deploy-cluster.sh [hunk 8]
```diff
@@ -83,10 +89,37 @@ aks-engine deploy --subscription-id ${SUBSCRIPTION_ID} \
   --client-secret ${CLIENT_SECRET}
 echo "Kubernetes cluster deployed. Please find the kubeconfig for it in _output/"
 
-export KUBECONFIG=_output/$(ls -t _output | head -n 1)/kubeconfig/kubeconfig.$LOCATION.json
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: why these two lines are removed?

File: hack/deploy-cluster.sh [hunk 9]
```diff
@@ -83,10 +89,37 @@ aks-engine deploy --subscription-id ${SUBSCRIPTION_ID} \
...
-echo "Kubernetes cluster deployed. Please find the kubeconfig at $KUBECONFIG"
 
-# Deploy AzureDisk CSI Plugin
+check=$(aks-engine version)
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: I think below changes are not needed, right? there is alreayd aks-engine deploy on above L82

Comment (Abhik1998) [reply hack/deploy-cluster.sh]: yes sir but previously, it was done

**Comment by feiskyer** [comment]
@Abhik1998 the commits history is broken. Please recreate the branch based on upstream/master.

**Pull request closed without merging**